controller/controller_client_test.go

```diff
@@ -21,70 +21,108 @@ import (
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
 	"github.com/elastic/elastic-agent-client/v7/pkg/client/mock"
 	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+	"github.com/elastic/elastic-agent-libs/logp"
 )
 
 func TestAgentControl(t *testing.T) {
-	unitOneID := mock.NewID()
+	unitOutputID := fmt.Sprintf("%s-output", mock.NewID())
+	unitInputID := fmt.Sprintf("%s-input", mock.NewID())
```

Comment:
This test should have multiple input units to match the current implementation in the agent.

You should additionally emulate:

1) Log level changes to individual units. The lowest log level of any unit should be the global log level of the shipper, until we create separate loggers for the input and output.
2) Removal and addition of individual input units to emulate a policy change that installs or removes a new integration that outputs tot he shipper.
3) Changing the configured output type, from Elasticsearch to console and back for example
4) Removal of all the input units.
5) Removal of just the output unit.
6) Updating the contents of the shared input unit configurations, for example changing the socket or port.
7) Changing an output parameter, for example changing the number of workers for the Elasticsearch output or enabling or disabling the `pretty` format of the console output.

That is a lot of tests, so likely you will need more than one case.

Before writing any of these unit tests I think you should start the shipper running under the agent, and actually make these policy edits and log + observe what the agent sends you. 

I don't think you are handling all of these event sequences properly right now (log levels in particular). You could ideally use the unit change sequences you log and then just encode them into a test case to ensure you are capturing what the agent actually does.